x-pack/test/api_integration/apis/endpoint/index_pattern.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect/expect.js';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  describe('Endpoint index pattern API', () => {
+    it('should retrieve the index pattern for events', async () => {
+      const { body } = await supertest.get('/api/endpoint/index_pattern/events').expect(200);
+      expect(body.indexPattern).to.eql('events-endpoint-*');
```

Comment:
❔ Given what we've seen in the last few days with index patterns in different parts of the app (ingest/be, fe, test genrators) falling out of sync with each other, I wonder if importing these from a shared place or typing them as TS literals  would be helpful

Reply:
Yeah that's a good point. The hope of this PR is to remove as many references to hard coded index patterns as possible. The index patterns should be constructed by the ingest manager for almost all scenarios. The reason I kept it as a local string here is because based on the current endpoint package the resulting index pattern show be exactly `events-endpoint-*` and I'm trying to prevent other portions of the app from importing that. If an index pattern is need, the app should instead use the index pattern retriever to get the index pattern at runtime.

One option we could do is have the resolver generator query kibana using the same api call that this test is doing to retrieve the index pattern. That would allow use to remove the hard coded instance in the generator but would require that kibana is up and running whenever we use the generator.